src/commands/dev/socket.rs

```diff
@@ -16,51 +20,78 @@ const KEEP_ALIVE_INTERVAL: u64 = 10;
-
-    // console.log messages are in the Runtime domain
-    // we must signal that we want to receive messages from the Runtime domain
-    // before they will be sent
-    let enable_runtime = protocol::runtime::SendMethod::Enable(1.into());
-    let enable_runtime = serde_json::to_string(&enable_runtime)?;
-    let enable_runtime = Message::Text(enable_runtime);
-    write.send(enable_runtime).await?;
-
-    // if left unattended, the preview service will kill the socket
-    // that emits console messages
-    // send a keep alive message every so often in the background
-    let (keep_alive_tx, keep_alive_rx) = mpsc::unbounded_channel();
-    tokio::spawn(keep_alive(keep_alive_tx));
-    let keep_alive_to_ws = keep_alive_rx.map(Ok).forward(write);
-
-    // parse every incoming message and print them
-    let print_ws_messages = {
-        read.for_each(|message| async {
-            let message = message.unwrap().into_text().unwrap();
-            log::info!("{}", message);
-            let message: Result<protocol::Runtime, failure::Error> = serde_json::from_str(&message)
-                .map_err(|e| failure::format_err!("this event could not be parsed:\n{}", e));
-            if let Ok(protocol::Runtime::Event(event)) = message {
-                println!("{}", event);
+    async move {
+        let socket_url = format!("wss://rawhttp.cloudflareworkers.com/inspect/{}", session_id);
+        let socket_url = Url::parse(&socket_url)?;
+
+        // we loop here so we can issue a reconnect when something
+        // goes wrong with the websocket connection
+        loop {
+            let (ws_stream, _) = connect_async(&socket_url)
+                .await
+                .expect("Failed to connect to devtools instance");
+
+            let (mut write, read) = ws_stream.split();
+
+            // console.log messages are in the Runtime domain
+            // we must signal that we want to receive messages from the Runtime domain
+            // before they will be sent
+            let enable_runtime = protocol::runtime::SendMethod::Enable(1.into());
+            let enable_runtime = serde_json::to_string(&enable_runtime)?;
+            let enable_runtime = Message::Text(enable_runtime);
+            write.send(enable_runtime).await?;
+
+            // if left unattended, the preview service will kill the socket
+            // that emits console messages
+            // send a keep alive message every so often in the background
+            let (keep_alive_tx, keep_alive_rx) = mpsc::unbounded_channel();
+
+            // every 10 seconds, send a keep alive message on the channel
+            let heartbeat = keep_alive(keep_alive_tx);
+
+            // when the keep alive channel receives a message from the
+            // heartbeat future, write it to the websocket
+            let keep_alive_to_ws = keep_alive_rx
+                .map(Ok)
+                .forward(write)
+                .map_err(|e| failure::format_err!("{:?}", e));
```

Comment:
This can also be mapped like so since it returns an error that implements `std::error::Error`:
```suggestion
                .map_err(Into::into);
```